x-pack/plugins/enterprise_search/public/applications/app_search/app_logic.ts

```diff
@@ -26,5 +33,93 @@ export const AppLogic = kea<MakeLogicType<IAppValues, IAppActions>>({
         initializeAppData: () => true,
       },
     ],
+    account: [
+      {} as IAccount,
+      {
+        initializeAppData: (_, { appSearch: account }) => account,
+        setOnboardingComplete: (state) => ({
+          ...state,
+          onboardingComplete: true,
+        }),
+      },
+    ],
+    configuredLimits: [
+      {} as IConfiguredLimits,
+      {
+        initializeAppData: (_, { configuredLimits }) => configuredLimits.appSearch,
+      },
+    ],
+    ilmEnabled: [
+      false,
+      {
+        initializeAppData: (_, { ilmEnabled }) => !!ilmEnabled,
+      },
+    ],
+  },
+  selectors: {
+    /**
+     * Massages the `role` data we receive from the Enterprise Search
+     * server into a more convenient format for front-end use
+     */
+    myRole: [
+      (selectors) => [selectors.account],
+      ({ role }) => {
+        if (!role) return {};
+
+        // Role ability function helpers
+        const myRole = {
+          can: (action: string, subject: string): boolean => {
+            return (
+              role?.ability?.manage?.includes(subject) ||
+              (Array.isArray(role.ability[action]) && role.ability[action].includes(subject))
+            );
+          },
+          // TODO: canHaveScopedEngines fn
```

Comment:
I'll add `canHaveScopedEngines` (one of the only other still-used role methods) in when we add the Role Mappings view. I left it out for now mostly because I was being lazy about writing a test for it 😂

Reply:
A comment I wrote on one of @JasonStoltz's PR (and maybe has already been discussed but I missed out on), I'm not sure the usefulness of leaving these TODOs in an open source repo available to the public.  This note would not be useful to someone outside of our team, maybe we should take it out of here and move to an issue instead.

Reply:
Even if it's only useful for our team, it's still useful - just because we're in an open source repo now doesn't necessarily mean every dev comment we write has to be useful to the every single dev in the world (just IMO). Obviously the exception to that being if we're referring to sensitive info that should remain internal, but I don't think `canHaveScopedEngines` applies in this case.

Also, while issues are great for tasks, documentation for references, and PRs for checking past context/didscussions, they add an extra level of friction required to understand why certain microdecisions were made from a code architecture point of view. Whereas a comment is immediately there, and in this case I think answers the question:

Why didn't I just declare `can` as its own separate const instead nesting it within `myRole.can`? The answer being: we expect other `canX` functions to be added at some point, so it makes sense to group them in a related obj. And if someone else comes across this comment between now and the point where we add `canHaveScopedEngines`, they'll have more context/understanding vs likely wanting to try to clean to this up 😛

Reply:
FWIW, if you search for TODO in this repo, there are 1084 results across 672 files 😲.

My opinion is that it is OK. A TODO with a little more context that would make sense to an outsider would certainly be more useful, but I don't see a TODO comment as an inherently bad thing.